servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -266,11 +262,20 @@ public CreatePathResult createPath(TachyonURI path, CreatePathOptions options)
 
     // Create the final path component. First we need to make sure that there isn't already a file
     // here with that name. If there is an existing file that is a directory and we're creating a
-    // directory, nothing needs to be done.
+    // directory, update persistence property of the directories if needed, otherwise, nothing needs
+    // to be done.
     Inode lastInode = currentInodeDirectory.getChild(name);
     if (lastInode != null) {
       if (lastInode.isDirectory() && options.isDirectory()) {
-        return new CreatePathResult();
+        if (!lastInode.isPersisted() && options.isPersisted()) {
+          lastInode.setPersisted(true);
+          traversalResult.getPersisted().add(lastInode);
+          String ufsPath = mMountTable.resolve(getPath(lastInode)).getPath();
+          UnderFileSystem ufs = UnderFileSystem.get(ufsPath, MasterContext.getConf());
+          ufs.mkdirs(ufsPath, false);
+        }
+        return new CreatePathResult(Collections.<Inode>emptyList(), createdInodes,
```

Comment:
I think it makes more sense to keep `List<Inode>` because in the future we could modify more than a single `Inode` and if we switch to `Inode` we would later need to switch back to `List<Inode>`. 

As for the return logic, the CreatePath logic would be easier to follow if we always return `modifiedInodes`, making sure that it only contains the inodes it should contain. In particular, the `modifiedInodes.add(currentInodeDirectory);` statement should only be executed if we actually set the last modification time for the existing inode. Makes sense?